app/views/contexts/calculated_summary_context.py

```diff
@@ -30,78 +35,101 @@ def build_groups_for_section(self, section, return_to_block_id):
                 self._metadata,
                 self._response_metadata,
                 self._schema,
-                location,
+                current_location,
                 self._language,
                 return_to="calculated-summary",
                 return_to_block_id=return_to_block_id,
             ).serialize()
             for group in section["groups"]
         ]
 
-    def build_view_context_for_calculated_summary(self, current_location):
-        block = self._schema.get_block(current_location.block_id)
-
-        return_to_block_id = block["id"]
+    def build_view_context_for_calculated_summary(
+        self, current_location: Location
+    ) -> Mapping:
+        # type ignores added as block will exist at this point
+        block_id: str = current_location.block_id  # type: ignore
+        block: ImmutableDict = self._schema.get_block(block_id)  # type: ignore
 
         calculated_section = self._build_calculated_summary_section(
             block, current_location
         )
         calculation = block["calculation"]
 
-        groups = self.build_groups_for_section(calculated_section, return_to_block_id)
+        groups = self.build_groups_for_section(
+            calculated_section, block_id, current_location
+        )
 
         formatted_total = self._get_formatted_total(
             groups or [],
             current_location=current_location,
-            calculation_operator=ValueSourceResolver.get_calculation_operator(
+            calculation=ValueSourceResolver.get_calculation_operator(
                 calculation["calculation_type"]
-            ),
+            )
+            if calculation.get("answers_to_calculate")
+            else calculation["operation"],
         )
 
-        context = {
+        collapsible = block.get("collapsible") or False
+        block_title = block.get("title")
+
+        return {
             "summary": {
                 "groups": groups,
                 "answers_are_editable": True,
                 "calculated_question": self._get_calculated_question(
                     calculation, formatted_total
                 ),
-                "title": block.get("title") % dict(total=formatted_total),
-                "collapsible": block.get("collapsible", False),
+                "title": block_title % dict(total=formatted_total)
+                if block_title
+                else None,
```

Comment:
Title is mandatory for calculated summary.

Reply:
Updated 👍